tests/test_exceptions_handler.py

```diff
@@ -220,7 +221,11 @@ def lookup(self, exception):
     with caplog.at_level(logging.WARNING):
         _, response = exception_handler_app.test_client.get("/1")
 
-    assert caplog.records[0].message == (
+    for record in caplog.records:
+        if record.message.startswith("You are"):
+            break
+
+    assert record.message == (
         "You are using a deprecated error handler. The lookup method should "
         "accept two positional parameters: (exception, route_name: "
         "Optional[str]). Until you upgrade your ErrorHandler.lookup, "
```

Comment:
Wouldn't this assertion have to be nested inside the `for` loop? Or does exiting the loop leave `record` defined as the last iterated item?

Reply:
Correct. It is not scoped. `record` is still set from the last iteration of the loop when it was broken.